core/drivers/firewall/stm32_rif.c

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2021-2024, STMicroelectronics
+ */
+
+#include <assert.h>
+#include <drivers/stm32_rif.h>
+#include <drivers/stm32mp_dt_bindings.h>
+#include <kernel/panic.h>
+#include <mm/core_memprot.h>
+
+static uint32_t get_scid_mask(unsigned int nb_cid_supp)
+{
+	uint32_t msb_nb_cid_supp = sizeof(nb_cid_supp) * 8 -
+				   __builtin_clz(nb_cid_supp | 1) - 1;
+
+	/* SCID bitfield highend can't be > SCID_SHIFT + MAX_CID_BITFIELD */
+	assert(msb_nb_cid_supp < MAX_CID_BITFIELD);
```

Comment:
Could you check this max max. It seems the max number of CIDs is 8 (ID 0 to 7), hence 3 bits are needed. Maybe replace `<` with `<=`.